### PR TITLE
Adding unix detection for ttyUSB/ttyACM ports

### PR DESCRIPTION
**Pull request by Gregwar**
This detects serial ports (keeping ttyUSB and ttyACM)
It is not a perfect method but it works in most case and still better than nothing anyway

### CODE REVIEW

File: SerialAPI.h [hunk 1]
```diff
@@ -87,6 +89,25 @@ class SerialAPI: public FB::JSAPIAuto
...
 		}
 #endif
+#ifdef __unix__
+            DIR *dev = opendir("/dev");
+            struct dirent *ent;
+
+            if (dev != NULL) {
+                while (ent = readdir(dev)) {
+                    std::string name(ent->d_name);
+                    std::string prefix = name.substr(0, 6);
```
Comment (billhsu) [inline SerialAPI.h]: @Gregwar what if the length of ```name``` is less then 6? an out_of_range exception will be thrown. Please a a check here. Something like below:
```C++
if (name.length()<=6) {
    continue;
}
```

File: SerialAPI.h [hunk 2]
```diff
@@ -87,6 +89,25 @@ class SerialAPI: public FB::JSAPIAuto
 			delete [] lpCC;
 		}
 #endif
+#ifdef __unix__
+            DIR *dev = opendir("/dev");
```
Comment (billhsu) [inline SerialAPI.h]: "/dev" should be moved to a constant.

**Comment by billhsu** [comment]
@Gregwar Thanks for submitting the PR! I have two minor comments.

**Comment by billhsu** [comment]
@Gregwar can you help take a look into #24? It seems that that PR is doing the same.

File: SerialAPI.h (hunk 1 above)
Comment (Gregwar) [inline SerialAPI.h]: From http://www.cplusplus.com/reference/string/string/substr/
"if the string is shorter, as many characters as possible are used"

**Pull request closed without merging**